gcc/config/arc/arc.md

```diff
@@ -3978,6 +4096,45 @@
    (set_attr "iscompact" "true")
    (set_attr "cond" "nocond")])
 
+(define_insn "jli_s"
+  [(match_operand:SI 0 "immediate_operand" "L")]
+  ""
```

Comment:
I think jli is only valid for ARCv2 processors. I miss this condition in your code.